packages/api/src/external/hie/set-doc-query-start.ts

```diff
@@ -0,0 +1,66 @@
+import { MedicalDataSource } from "@metriport/core/external/index";
+import { PatientExternalData } from "@metriport/core/domain//patient";
+import { Patient } from "@metriport/core/domain/patient";
+import { PatientModel } from "../../models/medical/patient";
+import { executeOnDBTx } from "../../models/transaction-wrapper";
+import { getPatientOrFail } from "../../command/medical/patient/get-patient";
+
+export type setDocQueryStartAt = {
+  patient: Pick<Patient, "id" | "cxId">;
+  source: MedicalDataSource;
+  startedAt: Date;
+};
+
+/**
+ * Set start time for document query progress
+ *
+ * @returns Updated patient
+ */
+export async function setDocQueryStartAt({
```

Comment:
Created a separate function from setDocQueryProgress to isolate the change to just updating the startedAt

Reply:
We should stop duplicating this logic over and over and come up with an update function for the entity that we can pass what's to be updated and it figures it out. Otherwise we'll end with 10's of duplicated logic for each permutation of fields.

This is somewhat related to https://github.com/metriport/metriport-internal/issues/1740

Reply:
So for the purposes of this PR should i just added a comment pointing to this PR?

Reply:
Sounds good, so we can get this out and we already have this in other places.

We just gotta address this at some point. I'd probably go as far as saying that we shouldn't create other variations of this - if needed, we should build in a way that's reusable and so we can use that ticket to start transitioning the other ones to the new one.